src/examples/using-with-cmake/host-config.cmake.in

```diff
@@ -29,5 +29,20 @@ set(MPI_Fortran_COMPILER   "@MPI_Fortran_COMPILER@"   CACHE PATH "")
 set(MPIEXEC_EXECUTABLE     "@MPIEXEC_EXECUTABLE@"     CACHE PATH "")
 set(MPIEXEC_NUMPROC_FLAG   "@MPIEXEC_NUMPROC_FLAG@"   CACHE STRING "")
 
+# CUDA
+set(ENABLE_CUDA                @ENABLE_CUDA@                  CACHE BOOL "")
+set(CUDA_TOOLKIT_ROOT_DIR      "@CUDA_TOOLKIT_ROOT_DIR@"      CACHE PATH "")
+set(CMAKE_CUDA_COMPILER        "@CMAKE_CUDA_COMPILER@"        CACHE PATH "")
+set(CMAKE_CUDA_HOST_COMPILER   "@CMAKE_CUDA_HOST_COMPILER@"   CACHE PATH "")
+set(CMAKE_CUDA_ARCHITECTURES   "@CMAKE_CUDA_ARCHITECTURES@"   CACHE STRING "")
+set(CMAKE_CUDA_FLAGS           "@CMAKE_CUDA_FLAGS@"           CACHE STRING "")
+
+# HIP
+set(ENABLE_HIP                @ENABLE_HIP@                  CACHE BOOL "")
+set(HIP_ROOT_DIR              "@HIP_ROOT_DIR@"              CACHE STRING "")
+set(HIP_CLANG_PATH            "@HIP_CLANG_PATH@"            CACHE STRING "")
+set(CMAKE_HIP_ARCHITECTURES   "@CMAKE_HIP_ARCHITECTURES@"   CACHE STRING "")
+set(CMAKE_EXE_LINKER_FLAGS    "@CMAKE_EXE_LINKER_FLAGS@"    CACHE STRING "")
```

Comment:
Similarly, should the `HIP`-related lines (other than the first) be wrapped in `if(ENABLE_HIP)` ?

Reply:
should these be #cmakedefines, so they only exist in the case where hip, etc are enabled? They are still a little clunky to use, but seems like it would avoid blank entries.

Reply:
Could go either way... I'd be fine with them guarded by an if... or do the fancier solution of printing the whole block in a cmake define like cyrus suggested.

Reply:
Went with if guards.